src/reanimated2/interpolation.js

```diff
@@ -1,30 +1,117 @@
+}
+
+function validateType(type) {
+  'worklet';
+
+  if (!type && type !== undefined) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      }) or interpolate(value, [inputRange], [outputRange], 'clamp')`
+    );
+  }
+
+  if (
+    type?.constructor === Object &&
+    ((Object.keys(type).length === 2 &&
+      !('extrapolateLeft' in type && 'extrapolateRight' in type)) ||
+      (Object.keys(type).length === 1 &&
+        !('extrapolateLeft' in type || 'extrapolateRight' in type)) ||
+      Object.keys(type).length > 2)
+  ) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      })`
+    );
+  }
+
+  if (typeof type === 'object') {
+    if ('extrapolateLeft' in type && !isExtrapolate(type.extrapolateLeft)) {
+      throw new Error(
+        `Reanimated: not supported value for "extrapolateLeft" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+         interpolate(value, [inputRange], [outputRange], {
+          extrapolateLeft: 'clamp',
+        })`
+      );
+    }
+
+    if ('extrapolateRight' in type && !isExtrapolate(type.extrapolateRight)) {
+      throw new Error(
+        `Reanimated: not supported value for "extrapolateRight" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+         interpolate(value, [inputRange], [outputRange], {
+          extrapolateRight: 'clamp',
+        })`
+      );
+    }
+  }
+
+  if (typeof type === 'string' && !isExtrapolate(type)) {
+    throw new Error(
+      `Reanimated: not supported value for "interpolate" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+       interpolate(value, [inputRange], [outputRange], "clamp")`
+    );
+  }
+}
+
+// TODO: support default values in worklets:
+// e.g. function interpolate(x, input, output, type = Extrapolate.CLAMP)
 function internalInterpolate(x, l, r, ll, rr, type) {
   'worklet';
   if (r - l === 0) return ll;
   const progress = (x - l) / (r - l);
   const val = ll + progress * (rr - ll);
   const coef = rr >= ll ? 1 : -1;
 
-  // TODO: support default values in worklets:
-  // e.g. function interplate(x, input, output, type = Extrapolate.CLAMP)
-  type = type || Extrapolate.EXTEND;
+  const config = { type, coef, val, ll, rr, x };
 
-  if (coef * val < coef * ll || coef * val > coef * rr) {
-    switch (type) {
-      case Extrapolate.IDENTITY:
-        return x;
-      case Extrapolate.CLAMP:
-        if (coef * val < coef * ll) {
-          return ll;
-        }
-        return rr;
-      case Extrapolate.EXTEND:
-      default:
-        return val;
+  validateType(type);
+
+  if (typeof type === 'object') {
```

Comment:
Just making sure, it can be null here? (`typeof null === "object"`)

Reply:
That's why I am doing `validateType` on `101` line

Reply:
Makes sense, thanks for explaining.

Line 101 if line numeration changes:
`validateType(type);`